crates/net/network/src/fetch/mod.rs

```diff
@@ -232,16 +242,31 @@ impl StateFetcher {
             ))
         }
 
-        if let Some(peer) = self.peers.get_mut(&peer_id) {
-            // If the peer is still ready to be accept new requests, we try to send a followup
-            // request immediately.
-            if peer.state.on_request_finished() {
-                return self.followup_request(peer_id)
+        if !is_likely_a_bad_message {
+            if let Some(peer) = self.peers.get_mut(&peer_id) {
+                // If the peer is still ready to be accept new requests, we try to send a followup
+                // request immediately.
+                if peer.state.on_request_finished() {
+                    return self.followup_request(peer_id)
+                }
             }
         }
+
         None
     }
 
+    fn is_likely_bad_message(request: &HeadersRequest, headers: &[Header]) -> bool {
```

Comment:
we will only use this on `PeerResult` so we can implement this for `PeerResult<Vec<Header>` directly, so we can call `result.is_likely_bad`